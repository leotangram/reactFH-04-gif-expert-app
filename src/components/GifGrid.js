import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem'

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category)

  return (
    <>
      <h3 className="animate__animated animate__bounceInLeft">{category}</h3>
      {loading && <p className="animate__animated flash">Cargando</p>}
      <div className="card-grid">
        {images.map(image => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string
}

export default GifGrid
