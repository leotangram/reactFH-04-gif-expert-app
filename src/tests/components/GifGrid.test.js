import React from 'react'
import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('GifGrid Component', () => {
  const category = 'One Punch'
  test('should show correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show items when images is loaded useFetchGif', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'hhtps://local.com',
        title: 'Potato'
      },
      {
        id: '123',
        url: 'hhtps://local.com',
        title: 'Potato'
      }
    ]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })
})
