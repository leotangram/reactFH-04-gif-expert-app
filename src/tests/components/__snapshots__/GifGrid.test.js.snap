// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Component should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid Component should show items when images is loaded useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInLeft"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Potato"
      url="hhtps://local.com"
    />
  </div>
</Fragment>
`;
